### DISCUSSION TITLE
What's the error rate of APPROX_COUNT_DISTINCT ?

### DISCUSSION THREAD

**Discussion by tarepanda1024** [apache/doris#6590] [A - General / Q&A] [answered]
I see the the APPROX_COUNT_DISTINCT and HLL_UNION_AGG are based on  HyperLogLog algorithm.I wan't to know what's the difference between APPROX_COUNT_DISTINCT and HLL_UNION_AGG ? And  what's the error rate of APPROX_COUNT_DISTINCT ?
(在文档中看到APPROX_COUNT_DISTINCT和HLL_UNION_AGG都是基于HyperLogLog算法来实现的，我想问一下它们的差别是什么？以及APPROX_COUNT_DISTINCT的精度是多少？可以调整吗)

**Comment by EmmyMiao87** [comment, marked as answer]
参数类型不一样，approx_count_distinct 参数类型没限制，但是 hll_union_agg 参数类型必须为 hll 类型。
实现上都用了hll 的算法，没差别。
误差率你可以 google 一下，hll 算法误差率，是有精确公式计算出来的

**Reply by tarepanda1024** [reply]
好的，看着HyperLogLog根据基数大小和桶大小的不同，精确度会不一样。在presto中可以设置精确度：
https://prestodb.io/docs/current/functions/aggregate.html?highlight=approx#id6
https://prestodb.io/docs/current/functions/hyperloglog.html?highlight=approx#id0
Doris中没有明确说明，看了一下hll.cpp，看着也有根据基数、空间大小进行调整的逻辑，不过感觉调用关系有点复杂，不是很懂，所以想请教一下。
